compiler/rustc_metadata/src/rmeta/decoder/cstore_impl.rs

```diff
@@ -501,7 +501,9 @@ pub(in crate::rmeta) fn provide(providers: &mut Providers) {
             tcx.arena
                 .alloc_slice(&CStore::from_tcx(tcx).crate_dependencies_in_postorder(LOCAL_CRATE))
         },
-        crates: |tcx, ()| tcx.arena.alloc_from_iter(CStore::from_tcx(tcx).crates_untracked()),
+        crates: |tcx, ()| {
+            tcx.arena.alloc_from_iter(CStore::from_tcx(tcx).iter_crate_data().map(|(cnum, _)| cnum))
```

Comment:
is this technically the only place where we need to freeze the `CStore` lock? If you can't iterate, it seems sound to still keep adding stuff to it.

Reply:
we should freeze it here either way to avoid it accidentally being called during name resolution.